test/e2e/tests/eg_uninstall_install.go

```diff
@@ -0,0 +1,107 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+//go:build e2e
+// +build e2e
+
+package tests
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+
+	"github.com/envoyproxy/gateway/internal/cmd/options"
+	"github.com/envoyproxy/gateway/internal/utils/helm"
+)
+
+func init() {
+	PackageTests = append(PackageTests, EGUninstallAndInstallTest)
+}
+
+var EGUninstallAndInstallTest = suite.ConformanceTest{
+	ShortName:   "EGUninstallAndInstall",
+	Description: "Uninstall and Install Envoy Gateway using Helm Package Tool",
+	Manifests:   []string{"testdata/eg-uninstall-install.yaml"},
+	Test: func(t *testing.T, suite *suite.ConformanceTestSuite) {
+		t.Run("Uninstall and Install Envoy Gateway should succeed", func(t *testing.T) {
```

Comment:
can we split this into two sub test run, one is uninstall and the other is install ?

Reply:
Let's imagine that if we use two sub-tests:
1. Install process: you need to uninstall the eg resource first, then install the eg resource, and finally test whether it is installed normally
2. Uninstall process: uninstall the eg resource directly. (How do we test the success of uninstallation?)

We can see that if we use two sub-tests, most of the processes are overlapped. I think we have more benefits by using a combined test

Reply:
What I means here is to split the whole test case into multiple `test.Run`, like first `t.Run("check svc xxx")`, then `t.Run("uninstall xxx")`, `t.Run("install xxx")`, not create a new test case for install/uninstall.

In this way, we can have a good understanding of the procedures in this test. It's technical feasible I think.

Reply:
Done!